dfdaemon/server.go

```diff
@@ -0,0 +1,122 @@
+package dfdaemon
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+
+	"github.com/dragonflyoss/Dragonfly/cmd/dfdaemon/app/options"
+	"github.com/dragonflyoss/Dragonfly/dfdaemon/config"
+	"github.com/dragonflyoss/Dragonfly/dfdaemon/handler"
+	"github.com/dragonflyoss/Dragonfly/dfdaemon/proxy"
+	"github.com/sirupsen/logrus"
+
+	"github.com/pkg/errors"
+)
+
+// Server represents the dfdaemon server
+type Server struct {
+	server *http.Server
+	proxy  *proxy.TransparentProxy
+}
+
+// Option is the functional option for creating a server
+type Option func(s *Server) error
+
+// WithTLSConfig sets the tls config for the server
+func WithTLSConfig(config *tls.Config) Option {
```

Comment:
It's unused in the whole project.

Reply:
I meant to use this in tests or for other more complex scenarios, but it doesn't seem very useful at the moment. I'll remove it for now